### PR TITLE
boards: shields: Add X-NUCLEO-WB05KN1 BLE shield

### PR DESCRIPTION
**Pull request by HoZHel**
Add support for X-NUCLEO-WB05KN1 which is a BLE expansion board based on the BLUENRG-LPS.

### CODE REVIEW

File: boards/shields/x_nucleo_wb05kn1/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+	depends on SHIELD_X_NUCLEO_WB05KN1_SPI
+
+config SPI_STM32_INTERRUPT
+	default y
+	depends on SPI
+
+choice BT_HCI_BUS_TYPE
+	default BT_SPI if DT_HAS_ST_HCI_SPI_V2_ENABLED
+	default BT_H4
+endchoice
```
Comment (msmttchr) [inline boards/shields/x_nucleo_wb05kn1/Kconfig.defconfig:19]: I think this needs to be updated due to changes in https://github.com/zephyrproject-rtos/zephyr/pull/72323 as BT_HCI_BUS_TYPE has been removed.

File: boards/shields/x_nucleo_wb05kn1/x_nucleo_wb05kn1_spi.overlay [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2024 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&arduino_spi {
+	cs-gpios = <&arduino_header 16 GPIO_ACTIVE_LOW>;	      /* D10 */
+
+	bluenrg-lp@0 {
```
Comment (msmttchr) [inline boards/shields/x_nucleo_wb05kn1/x_nucleo_wb05kn1_spi.overlay:10]: I think this needs to be updated following https://github.com/zephyrproject-rtos/zephyr/pull/72323.

**Review by msmttchr — CHANGES_REQUESTED** [review]
Some changes due to HCI driver changes introduced by https://github.com/zephyrproject-rtos/zephyr/pull/72323

File: boards/shields/x_nucleo_wb05kn1/Kconfig.defconfig (hunk 1 above)
Comment (HoZHel) [reply boards/shields/x_nucleo_wb05kn1/Kconfig.defconfig]: Done.

File: boards/shields/x_nucleo_wb05kn1/x_nucleo_wb05kn1_spi.overlay (hunk 2 above)
Comment (HoZHel) [reply boards/shields/x_nucleo_wb05kn1/x_nucleo_wb05kn1_spi.overlay]: Done.

**Review by msmttchr — APPROVED** [review]
Lgtm. Thanks

**Review by avisconti — DISMISSED** [review]
LGTM

**Review by kartben — CHANGES_REQUESTED** [review]

File: boards/shields/x_nucleo_wb05kn1/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
++----------------+-----------------------+
+
+More information about X-NUCLEO-WB05KN1 can be found here:
+       - `X-NUCLEO-WB05KN1 datasheet`_
+
+Programming
+***********
+
+Activate the presence of the shield for the project build by adding the
+``-DSHIELD=x_nucleo_wb05kn1_uart`` or ``-DSHIELD=x_nucleo_wb05kn1_spi`` arg to the build command
```
[the remark this answers is not in the document]
Comment (HoZHel) [reply boards/shields/x_nucleo_wb05kn1/doc/index.rst]: Done.

**Review by kartben — APPROVED** [review]

**Review by erwango — APPROVED** [review]

File: boards/shields/x_nucleo_wb05kn1/Kconfig.defconfig [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+if BT
+
+config SPI
+	default y
+	depends on SHIELD_X_NUCLEO_WB05KN1_SPI
+
+config SPI_STM32_INTERRUPT
+	default y
+	depends on SPI
```
Comment (jfischer-no) [inline boards/shields/x_nucleo_wb05kn1/Kconfig.defconfig:14]: Would this shield work on non-STM32 based boards?

Comment (HoZHel) [reply boards/shields/x_nucleo_wb05kn1/Kconfig.defconfig:14]: Yes, it works through SPI or UART. I have already tested on nRF52840-DK.

**Pull request merged**